pandas/_libs/tslibs/dtypes.pyx

```diff
@@ -77,7 +77,8 @@ _period_code_map = {
     "Q-OCT": 2010,    # Quarterly - October year end
     "Q-NOV": 2011,    # Quarterly - November year end
 
-    "M": 3000,        # Monthly
+    "M": 3000,        # Monthly - month end
+    "MS": 3001,       # Monthly - month start
```

Comment:
the integers here are semantically meaningful, 3001 doesnt make sense in this context

Reply:
OK, I'll try to obtain a better number. It would help if I knew the meaningful relation between "M" and 3000. If the last zero symbolises the end (of month), then by some convoluted logic maybe one could represent the beginning, though?

Reply:
If the idea is for "MS" to map to the same type of Period as "M", then just use 3000.  If the idea is to have a Period that specifically corresponds to "Start of the Month" rather than "Month", then that really isn't what Period represents.

Reply:
I don't think using 3000 for both "M" and "MS" would be correct - please have a look at this screenshot:
![image](https://user-images.githubusercontent.com/7954786/112280268-1f623180-8c85-11eb-84eb-042fb0566d3a.png)

Reply:
there are aliases already in _period_code_map, but they are added on L104 and L108 after _reverse_period_code_map is defined.

Reply:
Done.